samples/msal-node-samples/cachePlugin.ts

```diff
@@ -0,0 +1,62 @@
+import fs from "fs";
+import { TokenCacheContext, ICachePlugin } from "@azure/msal-node";
+
+const cachePluginFunctionality = (cacheLocation: string): ICachePlugin => {
+    const beforeCacheAccess = (
+        cacheContext: TokenCacheContext
+    ): Promise<void> => {
+        return new Promise((resolve, reject) => {
+            if (fs.existsSync(cacheLocation)) {
```

Comment:
I don't think a plaintext file is appropriate caching location for service tokens.

For app to app, just use memory. Bonus points for using an in-memory cache with eviction.

Reply:
Also, you should not cache ALL your tokens in the same location. Token cache lookup in MSAL NodeJS is O(n) (at least). 

For S2S, you need to partition the cache. Dogan wrote some helpers that shows how to do this for user, see distributedcacheplugin. https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-node/docs/caching.md#web-apps

Otherwise a multi-tenant app caching 1M tokens will just not work.

Reply:
This is just a sample. All other JS samples use a flat cache like this, and I believe the point is so that devs can see how the tokens are stored in the cache. I can remove if you'd like.

Reply:
Caching is good. Flat file cache not so much.